Khiem_Nguyen/rails/blogs_posts_messages_2/app/models/user.rb

```diff
@@ -0,0 +1,10 @@
+class User < ActiveRecord::Base
+    EMAIL_REGEX = /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]+)\z/i
+    has_many :comments, as: :commentable
+    has_many :posts
+    has_many :messages
+    has_many :owners
+    has_many :blogs, through: :owners
+    validates :first_name, :last_name, presence: true
```

Comment:
user has_many blogs through owners as well as through posts (Owner and post are the join table). //Refer to platform's more relationship tab and use source: :blog to establish user has_many blog_posts through posts